huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_flinkjar_job_test.go

```diff
@@ -0,0 +1,248 @@
+package dli
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/chnsz/golangsdk/openstack/dli/v1/flinkjob"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func getDliFlinkJarJobResourceFunc(config *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := config.DliV1Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmtp.Errorf("error creating Dli v1 client, err=%s", err)
+	}
+	jobId, _ := strconv.Atoi(state.Primary.ID)
+	return flinkjob.Get(client, jobId)
+}
+
+func TestAccResourceDliFlinkJarJob_basic(t *testing.T) {
+	var obj flinkjob.CreateJarJobOpts
+	resourceName := "huaweicloud_dli_flinkjar_job.test"
+	name := acceptance.RandomAccResourceName()
+	bucketName := acceptance.RandomAccResourceNameWithDash()
+
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&obj,
+		getDliFlinkJarJobResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckDliJarPath(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFlinkJarJobResource_basic(name, bucketName, acceptance.HW_REGION_NAME,
+					acceptance.HW_ACCESS_KEY, acceptance.HW_SECRET_KEY, acceptance.HW_DLI_FLINK_JAR_OBS_PATH),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "status", "job_running"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{},
+			},
+		},
+	})
+}
+
+func testAccFlinkJarJobResource_basic(name, bucketName, region, ak, sk, jarObsPath string) string {
+	return fmt.Sprintf(`
+variable "ak" {
+  type        = string
+  description = "value"
+  default     = "%s"
+
+}
+
+variable "sk" {
+  type        = string
+  description = "value"
+  default     = "%s"
+
+}
+
+variable "jarObsPath" {
+  type        = string
+  description = "value"
+  default     = "%s"
+}
+
+resource "huaweicloud_obs_bucket" "test" {
+  bucket        = "%s"
+  acl           = "private"
+  force_destroy = true
+}
+
+resource "huaweicloud_dli_package" "test" {
+  group_name  = "jarPackage"
+  type        = "jar"
+  object_path = var.jarObsPath
+}
+
+resource "huaweicloud_dli_queue" "test" {
+  name       = "%s"
+  cu_count   = 16
+  queue_type = "general"
+
+  tags = {
+    foo = "bar"
+  }
```

Comment:
[Suggestion] For non-test subjects, only need to declare the necessary parameters.